android/src/main/kotlin/com/kasem/receive_sharing_intent/ReceiveSharingIntentPlugin.kt

```diff
@@ -168,6 +174,27 @@ class ReceiveSharingIntentPlugin(val registrar: Registrar) :
             else -> null
         }
     }
+    private fun getPdfUris(context: Context, intent: Intent?): JSONArray? {
+        if (intent == null) return null
+
+        return when {
+            intent.action == Intent.ACTION_SEND -> {
+                val uri = intent.getParcelableExtra<Uri>(Intent.EXTRA_STREAM)
+                val path = FileDirectory.getAbsolutePath(context, uri)
+                if (path != null) {
+                    val type = getMediaType(path)
```

Comment:
This will conflict with other types of media for example video. Since you are adding a new media type `pdf`. You should also modify the media types returned to be more than 0 and 1. So the client know what type of media they are receiving. 

I suggest you do it like as follows.
```
0: Image
1: Videos
2: Pdf
```